src/main/java/seedu/address/model/animal/AnimalContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.animal;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.model.feedtime.FeedTime;
+import seedu.address.model.medicalcondition.MedicalCondition;
+
+/**
+ * Tests that any of an {@code Animal}'s fields matches any of the keywords given.
+ */
+public class AnimalContainsKeywordsPredicate implements Predicate<Animal> {
+    private final List<String> keywords;
+
+    public AnimalContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    /**
+     * Checks if the keyword matches any of the strings in an animal's feeding times.
+     * @param feedTimes A set containing all of the animal's feeding times.
+     * @param word A given keyword.
+     * @return the boolean that checks if the keyword matches.
+     */
+    public boolean containsFeedTime(Set<FeedTime> feedTimes, String word) {
+        if (!FeedTime.isValidFeedTime(word)) {
+            return false;
+        }
+
+        String concat = "";
+        for (FeedTime time : feedTimes) {
+            concat += time.feedTime + " ";
+        }
+        return StringUtil.containsWordIgnoreCase(concat, word);
+    }
+
+    /**
+     * Checks if the keyword matches any of the strings in an animal's medical conditions
+     * @param medicalConditions A set containing all of the animal's medical conditions.
+     * @param word A given keyword.
+     * @return the boolean that checks if the keyword matches.
+     */
+    public boolean containsMedicalCondition(Set<MedicalCondition> medicalConditions, String word) {
+        String concat = "";
+        for (MedicalCondition condition : medicalConditions) {
+            concat += condition.medicalConditionName + " ";
+        }
+        return StringUtil.containsWordIgnoreCase(concat, word);
+    }
+
+    @Override
+    public boolean test(Animal animal) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(animal.getName().fullName, keyword))
+                || keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(animal.getId().toString(), keyword))
+                || keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(animal.getSpecies().toString(), keyword))
+                || keywords.stream()
+                .anyMatch(keyword -> containsFeedTime(animal.getFeedTimes(), keyword))
+                || keywords.stream()
+                .anyMatch(keyword -> containsMedicalCondition(animal.getMedicalConditions(), keyword));
```

Comment:
nitpick: think the .anyMatch() methods from line 58 to 64 should have indented 8 more spaces since it is part of the stream on the line before it

Reply:
I think we should leave it as it is to be more consistent since lines 70-71 indicate the indentation that is used throughout the entire code base